### PR TITLE
Publish test splitter client as linux packages

### PR DESCRIPTION
**Pull request by nprizal**
### Description

Publish _stable_ version public Buildkite Packages repositories.
- ~[Alpine (apk)](https://buildkite.com/organizations/buildkite/packages/registries/test-splitter-apk)~ Skip for now, as there is issue publishing alpine package.
- [Debian (deb)](https://buildkite.com/organizations/buildkite/packages/registries/test-splitter-deb)
- [Red Hat (rpm)](https://buildkite.com/organizations/buildkite/packages/registries/test-splitter-rpm)

The packages are uploaded using `goreleaser` custom publisher.

### Verification
I have tested the pipeline, and this is the successful [build](https://buildkite.com/buildkite/test-splitter-client-release/builds/164#019105c8-4699-437a-a483-41ac6eb0d0a2). I have removed the published packages from that build.

### CODE REVIEW

File: .buildkite/steps/upload-linux-packages.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env sh
+
+file=${1}
```
Comment (nprizal) [inline .buildkite/steps/upload-linux-packages.sh:3]: this is the script that will be executed by `goreleaser` custom publisher

File: .goreleaser.yaml [hunk 2]
```diff
@@ -68,3 +68,28 @@ docker_manifests:
       - "buildkite/test-splitter:v{{ .Version }}-arm64"
     # skip pushing latest tag to Dockerhub if it's a prerelease  
     skip_push: auto
+
+nfpms:
```
Comment (nprizal) [inline .goreleaser.yaml:72]: `goreleaser` _pipe_ to build linux packages

File: .goreleaser.yaml [hunk 3]
```diff
@@ -68,3 +68,28 @@ docker_manifests:
...
+    formats:
+      - apk
+      - deb
+      - rpm
+    provides:
+      - test-splitter
+    release: 0
+
+publishers:
+  - name: buildkite-packages
```
Comment (nprizal) [inline .goreleaser.yaml:88]: `goreleaser` custom publisher to upload the linux packages to Buildkite Packages repository

**Review by wooly — APPROVED** [review]
Nice one @nprizal!

**Pull request merged**